### ISSUE TITLE
--sandbox_block_path should ignore missing directories

### ISSUE THREAD

**Issue by greggdonovan** [bazelbuild/bazel#4963]
### Description of the problem / feature request:

When I add the following to my `.bazelrc`, I get failures when `/search` does not exist:
```
build --sandbox_block_path=/search
test --sandbox_block_path=/search
```

### Feature requests: what underlying problem are you trying to solve with this feature?

I would like to exclude a directory from being readable in the sandbox to prevent likely cases of loss of hermeticity. If this directory does not exist, Bazel should ignore the `--sandbox_block_path` entry.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

> Replace this line with your answer.

### What operating system are you running Bazel on?

CentOS 7

### What's the output of `bazel info release`?

release 0.10.0

###  Have you found anything relevant by searching the web?

My [own StackOverflow question](https://stackoverflow.com/questions/43849651/how-to-lock-down-the-bazel-filesystem-sandbox). :)

**Comment by jin** [comment]
cc @philwo

**Comment by jmmv** [comment]
From my reading of the code, this is only an issue on Linux: we try to bind-mount these paths and their non-existence causes the failure. On macOS, we just use these paths to configure `sandbox-exec` and never try to access them so I think we don't fail there.

We should make the behavior consistent across platforms, document what this is in the flag description, and add a test.

**Comment by fmeum** [comment]
@bazel-io fork 7.3.0

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.3.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/7.3.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting `USE_BAZEL_VERSION=7.3.0rc1`. Thanks!
